server/src/actors/store.rs

```diff
@@ -0,0 +1,144 @@
+    producer: FutureProducer,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "snake_case")]
+enum KafkaMessageType {
+    Event,
+    AttachmentChunk,
+    Transaction,
+}
+
+#[derive(Debug, Serialize)]
+struct EventKafkaMessage {
+    ty: KafkaMessageType,
+    payload: Bytes,
+    start_time: i64,
+}
+
+impl StoreForwarder {
+    pub fn new(config: Arc<Config>) -> Self {
+        let mut client_config = ClientConfig::new();
+        for config_p in config.kafka_config() {
+            client_config.set(config_p.name.as_str(), config_p.value.as_str());
+        }
+
+        StoreForwarder {
+            config,
+            shutdown: SyncHandle::new(),
+            producer: client_config
+                .create()
+                .expect("could not create kafka producer"), // TODO cleanup
+        }
+    }
+}
+
+/// StoreMessageForwarder is an async actor since the only thing it does is put the messages
+/// in the kafka topics
+impl Actor for StoreForwarder {
+    type Context = Context<Self>;
+
+    fn started(&mut self, context: &mut Self::Context) {
+        log::info!("store forwarder started");
+        Controller::from_registry().do_send(Subscribe(context.address().recipient()));
+    }
+
+    fn stopped(&mut self, _ctx: &mut Self::Context) {
+        log::info!("store forwarder stopped");
+    }
+}
+
+/// Message sent to the StoreForwarder containing an event
+pub struct StoreEvent {
+    pub event_id: EventId,
+    pub payload: Bytes,
+}
+
+impl Message for StoreEvent {
+    type Result = Result<(), StoreError>;
+}
+
+fn serialize_kafka_event(payload: Bytes) -> Result<Vec<u8>, StoreError> {
+    let mut buff = Vec::new();
+    let message = EventKafkaMessage {
+        payload,
+        start_time: Utc::now().timestamp(),
+        ty: KafkaMessageType::Event,
+    };
+
+    match message.serialize(&mut Serializer::new(&mut buff)) {
+        Err(e) => Err(StoreError::SerializationError(e)),
+        _ => Ok(buff),
+    }
+}
+
+impl Handler<StoreEvent> for StoreForwarder {
+    type Result = ResponseFuture<(), StoreError>;
+
+    fn handle(&mut self, msg: StoreEvent, _ctx: &mut Self::Context) -> Self::Result {
+        let serialized_message = tryf!(serialize_kafka_event(msg.payload));
+
+        let record = FutureRecord::to(self.config.kafka_topic_name(KafkaTopic::Events))
+            .payload(&serialized_message)
+            .key(msg.event_id.0.as_bytes().as_ref());
+
+        let future = self.producer.send(record, 0).then(|result| match result {
+            Ok(Ok(_)) => Ok(()),
+            Ok(Err((kafka_error, _message))) => Err(StoreError::Kafka(kafka_error)),
+            Err(_) => Err(StoreError::Canceled),
+        });
+
+        Box::new(future)
+    }
+}
+
+impl Handler<Shutdown> for StoreForwarder {
+    type Result = ResponseFuture<(), TimeoutError>;
+
+    fn handle(&mut self, message: Shutdown, _context: &mut Self::Context) -> Self::Result {
+        self.producer.flush(message.timeout);
```

Comment:
I think we might need to move this into completion of the shutdown future. Essentially this:

```rust
let shutdown = match message.timeout {
    Some(timeout) => self.shutdown.timeout(timeout),
    None => self.shutdown.now(),
};

let future = shutdown.and_then(move |_| {
    self.producer.flush(message.timeout);
    Ok(())
});

Box::new(future)
```

That way we ensure that we first wait for all messages to be sent to Kafka, and then flush the producer.